packages/textInput/components/TextInput.tsx

```diff
@@ -75,12 +75,20 @@ export class TextInput<
     const labelContent = this.getLabelContent();
     const tooltipContent = this.getTooltipContent();
     const containerProps: { className?: string } = {};
+    const appearance = this.getInputAppearance();
+    const dataCy = [
+      "textInput",
+      ...(appearance && appearance !== InputAppearance.Standard
+        ? [`textInput.${appearance}`]
+        : [])
+    ].join(" ");
+
     if (this.props.className) {
       containerProps.className = this.props.className;
     }
     return (
-      <div {...containerProps}>
-        <div className={flex({ align: "center" })}>
+      <div {...containerProps} data-cy={dataCy}>
+        <div className={flex({ align: "center" })} data-cy="textInput-label">
           {labelContent}
```

Comment:
Is there a reason why we don't have a data-cy for the actual input here? I think it's the most used part of this component.

Reply:
That's a great point. I'll add it